src/app/services/views/ServiceTrayView.vue

```diff
@@ -0,0 +1,85 @@
+<template>
+  <RouteView name="service-summary-view">
+    <AppView>
+      <template #title>
+        <div class="tray-title-wrapper">
+          <img
+            aria-hidden="true"
+            src="@/assets/images/icon-wifi-tethering.svg?url"
+          >
+
+          <h2 class="tray-title">
+            <RouterLink
+              :to="{
+                name: 'service-detail-view',
+                params: {
+                  service: props.data.name,
+                },
+              }"
+            >
+              <RouteTitle
+                :title="t('services.routes.item.title', { name: props.data.name })"
+                :render="true"
+              />
+            </RouterLink>
+          </h2>
+        </div>
+      </template>
+
+      <div class="stack">
+        <div>
+          <h3>{{ t('services.overview') }}</h3>
+
+          <div class="mt-4">
+            <ExternalServiceDetails
+              v-if="props.data.serviceType === 'external'"
+              :mesh="props.data.mesh"
+              :service="props.data.name"
+            />
+
+            <ServiceInsightDetails
+              v-else
+              :service-insight="data"
+            />
+          </div>
+        </div>
+
+        <div v-if="props.data.serviceType === 'external'">
+          <h3>{{ t('services.config') }}</h3>
+
+          <ExternalServiceConfig
+            class="mt-4"
+            :mesh="props.data.mesh"
+            :service="props.data.name"
+          />
+        </div>
+      </div>
+    </AppView>
+  </RouteView>
+</template>
+
+<script lang="ts" setup>
+import ExternalServiceConfig from '../components/ExternalServiceConfig.vue'
+import ExternalServiceDetails from '../components/ExternalServiceDetails.vue'
+import ServiceInsightDetails from '../components/ServiceInsightDetails.vue'
+import { ServiceInsight } from '@/types/index.d'
+import { useI18n } from '@/utilities'
+
+const { t } = useI18n()
```

Comment:
We may aswell just use `<RouteView v-slot="{ t }" >` if possible? Its what we do everywhere else.

Reply:
See https://github.com/kumahq/kuma-gui/pull/1611#discussion_r1367041210.

Reply:
Honestly, I’m still confused why `t` is even provided on RouteView. Stuff like this should just be imported. I have no interest in warping these global components into some kind of import shorthand for utilities. It has no advantage other than saving the import.

Reply:
One reason is because it would be nice to be able to do `t('.introduction')` and not `t('services.routes.item.introduction')` (and that same thing repeated lots and lots of times in a route file).

If you don't use RouteView::t just in these files, that won't work here.

---

I've mentioned before that Vite's hot reloading works because it knows about imports, therefore if you change something in this file, it can reload everything else imported in this file, that can also then reload the TOKENs which are based on symbols, if the symbols change then that means you can sometimes get errors on a hot reload to do with the DI container not being able to find the right token (because you reloaded only a small part of the application). We mostly change route components (after all route components are mainly made up of pre-made reusable components that we edit rarely)

Again to note, this is only an issue whilst in development and this alone wouldn't lead me to want to do things like this. 

---

I've mentioned quite a few times before that it is good DX to make good choices easy and less good ones harder. Having `t` available in every single route means its more likely you will use it there and pass it into components via slots/attributes. We could potentially restrict this further by making it difficult to import `t` at all without a lint ignore via eslint (not saying I defo want to do that)

---

Separate to that, we do this everywhere else in the entire application in every single one of our RouteView's, why should this route take a different approach?

Reply:
I have to be honest, this is sounds very unconvincing. I’m strongly against preferring access of global utilities like i18n or can via slots instead of imports. The latter are vastly simpler and more transparent and there is no reason to impose such a restriction.

I understand that this makes sense if you want to automatically handle something like automatic i18n key trimming, but that sounds unnecessarily complex and not very useful.

Imports are the right tool for this job.

Reply:
They aren't global. The below is pretty much what we are doing and I don't see a problem with this approach. Its what we in do in more "traditional JS" for all our mocks also

```javascript
import RouterView from 'RouterView.vue'
import RouteView from 'RouteView.vue'
RouterView({}, {} => {
  RouteView({ name: 'service-list-view' }, ({can, t, route}) => {
    if(can('use routes') && route.name) {
        return t('services.routes.label', {name})
    }
  })
})
```

Passing the 'tooling' you need specific to the thing you are using as an argument of a callback in JS is an extremely common pattern, not only in our application but in the wider ecosystem. It's one approach to _avoid_ using globals.

Here's the same approach in all our mocks:

```javascript
export default ({env, fake}) => {
   return (params) => {
       return {
          name: params.name,
          createdAt: fake.date()
       }
   }
}
```

---

The reason to not use `t` or `can` inside reusable components, is that passing the data down into the reusable component instead makes it far more reusable.

We discussed this reasoning when we started adding out i18n support a while ago, and we said by default we would use attrs or slots to pass i18n text into components, but it would also be fine to also use `t` inside a reusable component where it made sense. I think this was written into the MADR for that. I think this argument is also very similar for the `h*` discussion we had on this PR somewhere. Its far better to not use `h*` elements inside reusable components as it makes the level incorrect when used in certain places i.e. it makes them less reusable.

---

I've given a good few reasons and advantages with examples and use cases now why we do things like this in this circumstance. I've also explained how this pattern is extremely common, not only in our application but in the wider ecosystem.

As far as I understand it at the moment, your argument for not using them is that you have a personal preference here to use imports. You say things like "its vastly simpler and more transparent" and "there is no reason to impose restrictions" without backing those statements up with any facts.

Considering that currently we use heavily use this approach everywhere, I think if you want to change how we access these utilities, then it would be good to put together/document a convincing argument outside of this PR.

If that argument is convincing enough and you can show its more advantageous then I'd happily change approach. But all that should happen outside of this PR, and I don't think we should change our current approach just in this instance and just in the code added in this PR.

Reply:
@kleinfreund I've just unresolved this again. Let me know.

Reply:
> They aren't global.

I mean this in the sense that they're globally-available utilities. Utilities that can be used everywhere unconditionally. Not actual global `window` properties. And they should be globally available. We can use them virtually everywhere in our application. That's a good thing.

> The below is pretty much what we are doing and I don't see a problem with this approach.

It works, sure, but that doesn't mean we should rule out using imports. Imports are by far the preferable mechanism to gain access to things in JavaScript. I fail to see how most of your arguments actually negate that that’s appropriate.

> It's one approach to _avoid_ using globals.

This isn’t about globals. And we shouldn't avoid using those utilities.

> The reason to not use `t` or `can` inside reusable components, is that passing the data down into the reusable component instead makes it far more reusable.
> 
> We discussed this reasoning when we started adding out i18n support a while ago, and we said by default we would use attrs or slots to pass i18n text into components, but it would also be fine to also use `t` inside a reusable component where it made sense. I think this was written into the MADR for that. I think this argument is also very similar for the `h*` discussion we had on this PR somewhere. Its far better to not use `h*` elements inside reusable components as it makes the level incorrect when used in certain places i.e. it makes them less reusable.

I don't understands how that relates to gaining access to i18n/can in the first place. In some components, we will need to get access to i18n/can so that we can pass its results down to "smaller" components. The latter part is not relevant to this conversation. Only the former is. And the former should be accomplished using imports, not `RouteView` slot properties.

> I've given a good few reasons and advantages with examples and use cases now why we do things like this in this circumstance. I've also explained how this pattern is extremely common, not only in our application but in the wider ecosystem.
> 
> As far as I understand it at the moment, your argument for not using them is that you have a personal preference here to use imports. You say things like "its vastly simpler and more transparent" and "there is no reason to impose restrictions" without backing those statements up with any facts.
> 
> Considering that currently we use heavily use this approach everywhere, I think if you want to change how we access these utilities, then it would be good to put together/document a convincing argument outside of this PR.
> 
> If that argument is convincing enough and you can show its more advantageous then I'd happily change approach. But all that should happen outside of this PR, and I don't think we should change our current approach just in this instance and just in the code added in this PR.

I'm not sure why I need to provide an argument for why using JavaScript imports is a good idea. That's just the best mechanism to get access to things in JavaScript we have. It’s tooling and framework-independent and everyone who knows JavaScript knows how to use imports.

Abstracting away importing things into symbols accessed via scoped slots of Vue components we **happen** to use a lot is unnecessary.

Just because we already use this approach for accessing i18n/can doesn't mean it's good or better. As it stands, it's roughly 50/50 between using `useI18n()` from an import and from RouteView. You may also recall that I was not in favor adding it there in the first place because I didn't see and still don't see the point in doing so.

So my argument boils down to "we really don’t need the extra abstraction" and I'm not sure how I can substantiate that with more examples.

More to the point: I don't want to spent time on the suggestion to not use imports for a situation where they’re the right tool.

I’m resolving this thread again because I don’t want to accept the suggestion.

Reply:
> Just because we already use this approach for accessing i18n/can doesn't mean it's good or better. As it stands, it's roughly 50/50 between using useI18n() from an import and from RouteView.

I just went through all our route components, I counted about 3 or 4 that use `import` all the rest (I dunno 50 odd) use `RouteView::t`.

If you want to change this approach, please can you give some reasoning/facts instead of "I don't like it / I'm not in favour / we don't need to", and I'll quite happily reconsider. Outside of this PR.

At the moment this discussion is getting in the way of us progressing here, whereas if you said "ok I'll put something together and we can discuss separately, if we decide the other way we can change the entire application in one go", then it wouldn't be in the way of us progressing, and importantly we wouldn't be ignoring you concerns. We would be prioritising well.

At the moment we can't get this PR through because we have painted the entire house red, and you want to paint one window frame blue. We can't finish the job here until you agree to use red instead of blue. We can always paint over the red with blue entirely at a later date.

---

Just to point out, we don't access t via an `import` we access its getter/token which we access via an import. Then we have to call the getter. Thats two steps, instead of no steps (I'm using RouteView anyway so there are no extra steps)

```
import { useI18n } from '@/utilities'
const { t } = useI18n()
```

Reply:
I’m not changing any rule-encoded approach or well-established pattern here. There is no rule that says you can’t use imports for i18n when RouteView is in the component.

The reasons for why you want that aren’t convincing me:

You say it’s what we do everywhere. That’s not true in general and in the case of where we use RouteView it’s down to you preferring this and writing it like this and not accepting when I do it differently.

You mention the Vite HMR issue, but as you said, this shouldn’t influence our decision making in general.

You mention that you have ideas around stripping i18n keys in certain places to avoid the repetitive prefixes, but that’s not something we do today and my PRs shouldn’t be the place where I need to do the pre-emptive refactoring work for something we may or may not implement.

Lastly, you talk about how it’s good DX to make good choices easy, and bad ones hard, etc. Yes, but I just disagree that this applies here. If anything, I think it’s a bad choice to restrict access to i18n in the way you mention you want to. We need i18n in a ton of places and enforcing a rule where we need to always pass i18n values down from a route component sounds like bad DX with hardly any active benefits.

**If** we implement something like i18n key stripping leveraging RouteView, **then** we can discuss whether such a rule makes sense.

**If** we agree that we need to enforce not being able to import i18n in components, **then** we can stop using those imports.

Since I’m not actually changing an established rule here, I treat this as a suggestion and I’m not accepting it because, yes, I _prefer_ using the import style.

Reply:
> When writing components we should "default to" avoiding using `t` within
generic/re-usable components. i.e. we should pass translated text into the
components from the outside using slots as much as possible (and occasionally
attributes). Mostly this means `t` would most commonly be used in route/view
components.

That was agreed upon and approved ^

> **If** we agree that we need to enforce not being able to import i18n in components, then we can stop using those imports.

i think this means you should stop using these imports here.

---

> You mention the Vite HMR issue, but as you said, this shouldn’t influence our decision making in general.

I said "that alone"

Reply:
> > When writing components we should "default to" avoiding using `t` within
> > generic/re-usable components. i.e. we should pass translated text into the
> > components from the outside using slots as much as possible (and occasionally
> > attributes). Mostly this means `t` would most commonly be used in route/view
> > components.
> 
> That was agreed upon and approved ^

That rule is about generic components and doesn’t apply here because we’re talking about a route component. It also doesn’t necessitate or imply that we shouldn’t use i18n in components other than route components and generic components (i.e. those that fall somewhere in between those categories).

> > **If** we agree that we need to enforce not being able to import i18n in components, then we can stop using those imports.
> 
> i think this means you should stop using these imports here.

But we didn’t. We agreed that should default to avoiding them in _generic_ components.

Reply:
Ok we spoke about this offline and said lets move forwards with this as is. I'm super happy to resolve now, gimme 2 tics I should be able to approve also.